sql/src/main/java/org/apache/druid/sql/calcite/parser/ExternalDestinationSqlIdentifier.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.sql.calcite.parser;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.calcite.sql.SqlIdentifier;
+import org.apache.calcite.sql.SqlNode;
+import org.apache.calcite.sql.SqlWriter;
+import org.apache.calcite.sql.parser.SqlParserPos;
+import org.apache.druid.error.DruidException;
+import org.apache.druid.java.util.common.StringUtils;
+import org.apache.druid.storage.StorageConnectorProvider;
+import org.apache.druid.utils.CollectionUtils;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Extends the {@link SqlIdentifier} to hold parameters for an external destination.
+ */
+public class ExternalDestinationSqlIdentifier extends SqlIdentifier
+{
+  private final Map<String, String> properties;
+
+  public ExternalDestinationSqlIdentifier(
+      String name,
+      SqlParserPos pos,
+      Map<String, String> properties
+  )
+  {
+    super(name, pos);
+    this.properties = properties;
+  }
+
+  public String getDestinationType()
```

Comment:
Since this is being used for the labels of permissions, please confirm that valid names here line up with the equivalent input source types returned from `InputSource#getTypes`. That way, permission labels will be consistent for reading and writing.

Reply:
Changed this to directly refer to the input source types directly.